### PR TITLE
remove deplicate code for PodRequestsAndLimits

### PR DESCRIPTION
**Pull request by yanxuean**
remove deplicate code for PodRequestsAndLimits
add unit test for PodRequestsAndLimits 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yanxuean** [comment]
/cc @xiangpengzhao 
/assign @liggitt 
PTAL

File: pkg/api/resource/helpers_test.go [hunk 1]
```diff
@@ -17,9 +17,11 @@ limitations under the License.
 package resource
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
```
Comment (dashpole) [inline pkg/api/resource/helpers_test.go]: I think we generally avoid using assert.

Comment (yanxuean) [reply pkg/api/resource/helpers_test.go]: Can you tell me the reason? I see that there are plenty of using assert in kubernetes now. :)

Comment (dashpole) [reply pkg/api/resource/helpers_test.go]: Yes, it isnt enforced in any way.  See https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures.
In general, it is better to use the primitives that go provides rather than using a bunch of different libraries across the repo for testing, especially if you are essentially just checking equality.

Comment (yujuhong) [reply pkg/api/resource/helpers_test.go]: The guideline only mentioned that the failure message should include useful error messages. I don't think it says to not use an assert library.

I'll admit it. I've used the library a lot. It makes the tests more concise and readable and does not provide less information. Is there any issue in our repository discussing this?

Comment (dashpole) [reply pkg/api/resource/helpers_test.go]: I dont believe so. I got the comment in a different code review. I guess its fine to use it here then

Comment (yanxuean) [reply pkg/api/resource/helpers_test.go]: OK, I will keep it.

**Comment by yanxuean** [comment]
@liggitt PTAL

**Comment by liggitt** [comment]
sorry, this didn't show up in my queries due to the required rebase... if this is still active, please rebase it, get tests green, and reassign me for approval once it has lgtm

/unassign

**Comment by dashpole** [comment]
the changes look reasonable, but I don't see any new unit tests?

**Comment by yanxuean** [comment]
@dashpole  The other guy have added some similar unit test cases in https://github.com/kubernetes/kubernetes/pull/60967.  so I remove mine. I am too late:)

**Comment by yanxuean** [comment]
@liggitt PTAL

**Comment by yanxuean** [comment]
/assign @liggitt 
PTAL

File: pkg/api/resource/helpers.go [hunk 2]
```diff
@@ -28,47 +28,41 @@ import (
...
-			if value, ok := limits[name]; !ok {
-				limits[name] = *quantity.Copy()
+	add := func(list map[api.ResourceName]resource.Quantity, containerResList api.ResourceList) map[api.ResourceName]resource.Quantity {
+		for name, quantity := range containerResList {
+			if value, ok := list[name]; !ok {
+				list[name] = *quantity.Copy()
 			} else {
 				value.Add(quantity)
-				limits[name] = value
+				list[name] = value
```
Comment (liggitt) [inline pkg/api/resource/helpers.go]: if this is mutating `list` in place, why do we also need to return it? that's makes this a little hard to follow.

also, could `add` and `max` be separate private functions? making them closures makes it hard to follow if they depend on variables from the PodRequestsAndLimits scope

File: pkg/api/resource/helpers.go [hunk 3]
```diff
@@ -28,47 +28,41 @@ import (
...
-				limits[name] = *quantity.Copy()
+				list[name] = *quantity.Copy()
 				continue
 			}
 			if quantity.Cmp(value) > 0 {
-				limits[name] = *quantity.Copy()
+				list[name] = *quantity.Copy()
 			}
 		}
+		return list
```
Comment (liggitt) [inline pkg/api/resource/helpers.go]: same question about modifying `list` in place and needing to return it

File: pkg/api/resource/helpers.go (hunk 2 above)
Comment (yanxuean) [reply pkg/api/resource/helpers.go]: The function do be better.  Done. Thanks

File: pkg/api/resource/helpers.go [hunk 4]
```diff
@@ -25,50 +25,41 @@ import (
...
-	reqs, limits = map[api.ResourceName]resource.Quantity{}, map[api.ResourceName]resource.Quantity{}
-	for _, container := range pod.Spec.Containers {
-		for name, quantity := range container.Resources.Requests {
-			if value, ok := reqs[name]; !ok {
-				reqs[name] = *quantity.Copy()
-			} else {
-				value.Add(quantity)
-				reqs[name] = value
-			}
+func addResourceList(list, new api.ResourceList) {
```
Comment (liggitt) [inline pkg/api/resource/helpers.go]: needs a little doc of what it does ("adds the resources in newList to list")

File: pkg/api/resource/helpers.go [hunk 5]
```diff
@@ -25,50 +25,41 @@ import (
...
-		for name, quantity := range container.Resources.Limits {
-			if value, ok := limits[name]; !ok {
-				limits[name] = *quantity.Copy()
-			} else {
-				value.Add(quantity)
-				limits[name] = value
+	}
+}
+
+func maxResourceList(list, new api.ResourceList) {
```
Comment (liggitt) [inline pkg/api/resource/helpers.go]: needs a little doc ("sets list to the greater of list/newList for every resource either list")

File: pkg/api/resource/helpers.go [hunk 6]
```diff
@@ -25,50 +25,41 @@ import (
...
+			if quantity.Cmp(value) > 0 {
+				list[name] = *quantity.Copy()
 			}
 		}
 	}
+}
+
+// PodRequestsAndLimits returns a dictionary of all defined resources summed up for all
+// containers of the pod.
+func PodRequestsAndLimits(pod *api.Pod) (reqs api.ResourceList, limits api.ResourceList) {
```
Comment (liggitt) [inline pkg/api/resource/helpers.go]: you still need the initalization step to make the targets non-nil:
```
reqs, limits = map[api.ResourceName]resource.Quantity{}, map[api.ResourceName]resource.Quantity{}
```

Comment (yanxuean) [reply pkg/api/resource/helpers.go]: thanks, done
